### PR TITLE
CLN: Assorted cleanups

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/core/algorithms.py [hunk 1]
```diff
@@ -743,7 +742,7 @@ def value_counts(
 
     else:
 
-        if is_extension_array_dtype(values) or is_sparse(values):
+        if is_extension_array_dtype(values):
```
Comment (gfyoung) [inline pandas/core/algorithms.py]: For everyone's edification: why can we remove this check?

File: pandas/core/construction.py [hunk 2]
```diff
@@ -528,11 +536,13 @@ def _try_cast(arr, dtype, copy, raise_cast_failure):
         if is_categorical_dtype(dtype):
             # We *do* allow casting to categorical, since we know
             # that Categorical is the only array type for 'category'.
+            dtype = cast(CategoricalDtype, dtype)
             subarr = dtype.construct_array_type()(
                 arr, dtype.categories, ordered=dtype._ordered
             )
         elif is_extension_array_dtype(dtype):
             # create an extension array from its dtype
+            dtype = cast(ExtensionDtype, dtype)
```
Comment (gfyoung) [inline pandas/core/construction.py]: For everyone's edification: why are we doing this?  Same for the similar cast a few lines above.

File: pandas/core/algorithms.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/algorithms.py]: is_sparse is a subset of is_extension_array_dtype

File: pandas/core/construction.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/construction.py]: This is to keep mypy from complaining

**Comment by jreback** [comment]
thanks @jbrockmendel

**Pull request merged**